### PR TITLE
login-auth

### PR DESCRIPTION
**Pull request by arkapaal**
please approve the changes 
i have added the login authentication feature,if a user has already logged in then it will not show the login button 
reference: https://medium.com/@dennisivy/creating-protected-routes-with-react-router-v6-2c4bbaf7bc1c

Fixes # (issue) *(only if applicable)*

### Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other (please specify):

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes

### Screenshots (if applicable)

**Please include any relevant screenshots or images that help explain the changes made.**
Screenshot 2024-10-20 001413
Screenshot 2024-10-20 001428

#### Additional Information 
- Name: Arka Pal
- Email:2305523@kiit.ac.in
- Roll No:2305523
- Branch: CSE

### CODE REVIEW

**Comment by arkapaal** [comment]
please approve the changes and also point out the mistakes

File: src/components/PrivateRoute.jsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import {Route,Navigate, Outlet} from 'react-router-dom'
+const PrivateRoute=({})=>{
+    let auth={'token':false}
```
Comment (dead8309) [inline src/components/PrivateRoute.jsx:3]: Just add this comment above this this 
`TODO: Replace with actual auth logic`

Comment (arkapaal) [reply src/components/PrivateRoute.jsx:4]: done the required changes 
please approve

**Comment by Tony-ArtZ** [comment]
Good work @arkapaal

**Pull request merged**